contrib/endpoints/src/api_manager/auth/service_account_token.cc

```diff
@@ -59,11 +59,12 @@ void ServiceAccountToken::SetAudience(JWT_TOKEN_TYPE type,
   jwt_tokens_[type].set_audience(audience);
 }
 
-const std::string& ServiceAccountToken::GetAuthToken(JWT_TOKEN_TYPE type) {
+const std::string& ServiceAccountToken::GetAuthToken(JWT_TOKEN_TYPE type,
+                                                     bool ignore_cache) {
   // Uses authentication secret if available.
   if (!client_auth_secret_.empty()) {
     GOOGLE_CHECK(type >= 0 && type < JWT_TOKEN_TYPE_MAX);
-    if (!jwt_tokens_[type].is_valid(0)) {
+    if (ignore_cache || !jwt_tokens_[type].is_valid(0)) {
```

Comment:
It is better to compare the new audience with old audience,  if diff, wipe out the cache.